java/src/jni/h5util.c

```diff
@@ -4477,91 +4911,36 @@ translate_wbuf(JNIEnv *env, jobjectArray in_buf, jlong mem_type_id, H5T_class_t
-                        }
-                        break;
-                    }
-                    case sizeof(jshort): {
-                        jshort shortValue = ENVPTR->CallShortMethod(ENVONLY, jobj, shortValueMid);
-                        for (x = 0; x < typeSize; x++) {
-                            ((char *)raw_buf)[i * typeSize + x] = ((char *)&shortValue)[x];
-                        }
-                        break;
-                    }
-                    case sizeof(jint): {
-                        jint intValue = ENVPTR->CallIntMethod(ENVONLY, jobj, intValueMid);
-                        for (x = 0; x < typeSize; x++) {
-                            ((char *)raw_buf)[i * typeSize + x] = ((char *)&intValue)[x];
-                        }
-                        break;
-                    }
-                    case sizeof(jlong): {
-                        jlong longValue = ENVPTR->CallLongMethod(ENVONLY, jobj, longValueMid);
-                        for (x = 0; x < typeSize; x++) {
-                            ((char *)raw_buf)[i * typeSize + x] = ((char *)&longValue)[x];
-                        }
-                        break;
-                    }
-                }
-                ENVPTR->DeleteLocalRef(ENVONLY, jobj);
-            }
-            break;
-        } /* H5T_INTEGER */
-        case H5T_FLOAT: {
+        case H5T_ENUM:
+        case H5T_BITFIELD:
+        case H5T_OPAQUE:
+        case H5T_INTEGER:
+        case H5T_FLOAT:
+        case H5T_REFERENCE: {
             /* Convert each list to an array element */
             for (i = 0; i < (size_t)count; i++) {
                 if (NULL == (jobj = ENVPTR->GetObjectArrayElement(ENVONLY, (jobjectArray)in_buf, (jsize)i)))
                     CHECK_JNI_EXCEPTION(ENVONLY, JNI_FALSE);
-                switch (typeSize) {
-                    case sizeof(jfloat): {
-                        jfloat floatValue = ENVPTR->CallFloatMethod(ENVONLY, jobj, floatValueMid);
-                        for (x = 0; x < typeSize; x++) {
-                            ((char *)raw_buf)[i * typeSize + x] = ((char *)&floatValue)[x];
-                        }
-                        break;
-                    }
-                    case sizeof(jdouble): {
-                        jdouble doubleValue = ENVPTR->CallDoubleMethod(ENVONLY, jobj, doubleValueMid);
-                        for (x = 0; x < typeSize; x++) {
-                            ((char *)raw_buf)[i * typeSize + x] = ((char *)&doubleValue)[x];
-                        }
-                        break;
-                    }
-                }
+                translate_atomic_wbuf(ENVONLY, jobj, mem_type_id, type_class,
+                                      ((char *)raw_buf) + i * typeSize);
                 ENVPTR->DeleteLocalRef(ENVONLY, jobj);
             }
             break;
-        } /* H5T_FLOAT */
-        case H5T_REFERENCE: {
+        }
+        case H5T_STRING: {
```

Comment:
The implementation below for converting strings looks the same as for the group of types above. I'm guessing it can just use the same logic in the way that translate_rbuf includes H5T_STRING among the types that do the same thing for conversion?

Reply:
I was waiting for the fix for variable strings - done.